manifests/resource/location.pp

```diff
@@ -153,66 +153,65 @@
 #  }
 
 define nginx::resource::location (
-  $ensure                      = present,
-  $internal                    = false,
-  $location                    = $name,
-  $server                      = undef,
-  $www_root                    = undef,
-  $autoindex                   = undef,
-  $index_files                 = [
+  Enum['present', 'absent'] $ensure                   = present,
+  Boolean $internal                                   = false,
+  String $location                                    = $name,
+  String $server                                      = undef,
```

Comment:
I'm not sure what the best practice is. in the other PRs we tried to always default to undef on required params without default value